net/tcp/tcp_send_unbuffered.c

```diff
@@ -348,6 +348,10 @@ static uint16_t tcpsend_eventhandler(FAR struct net_driver_s *dev,
                  &pstate->snd_buffer[pstate->snd_acked],
                  sndlen);
 
+      /* Notify the device driver of the availability of TX data */
+
+      tcp_send_txnotify(psock, conn);
```

Comment:
I think the top level will call tcp_send_txnotify, why we need call here?

Reply:
Yes, I did a wrong thing. Please, disregard that first commit. I have totally reworked the patch.